File: 2020/csharp/Aoc2020.Tests/Day09Tests.cs
using System.Linq;
using System.Collections.Generic;
using Xunit;

namespace Aoc2020.Tests
{
    public class Day09Tests
    {
        [Fact]
        public void FindFirstInvalidNumberWorks()
        {
            var data = Utils.ReadInputLines("09-test").Select(long.Parse).ToArray();
            Assert.Equal(127, Day09.FindFirstInvalidNumber(data, 5));
        }

        [Fact]
        public void FindWeaknessNumberWorks()
        {
            var data = Utils.ReadInputLines("09-test").Select(long.Parse).ToArray();
            Assert.Equal(62, Day09.FindWeakness(data, 127));
        }

        [Fact]
        public void Stars()
        {
            var run = new Day09();
            Assert.Equal("167829540", run.Result1());
            Assert.Equal("28045630", run.Result2());
        }
    }
}

File: 2022/fsharp/Aoc2022.Tests/Day09.Tests.fs
module Day09.Tests

open Xunit
open Utils
open Day09

let input =
    """
    R 4
    U 4
    L 3
    D 1
    R 4
    D 1
    L 5
    R 2
    """
    |> String.toLines

let input2 =
    """
    R 5
    U 8
    L 8
    D 3
    R 17
    D 10
    L 25
    U 20
    """
    |> String.toLines

[<Fact>]
let ``Day09 countTailPositions works`` () =
    Assert.Equal(13, input |> countTailPositions 2)
    Assert.Equal(1, input |> countTailPositions 10)
    Assert.Equal(36, input2 |> countTailPositions 10)

[<Fact>]
let ``Day09 Stars`` () =
    try
        Assert.Equal("6098", job1 ())
        Assert.Equal("2597", job2 ())
    with :? System.NotImplementedException ->
        ()
